### PR TITLE
Currency command

### PR DESCRIPTION
**Pull request by Dreary**
Currency command to use in-game under the "player" option

Example:
/player currency [currency type] [amount]

## Summary by CodeRabbit

- **New Features**
  - Introduced a `CurrencyCommand` allowing players to add various types of currency to their accounts.
- **Improvements**
  - Enhanced error handling for invalid currency types for better user experience.
  - Improved parsing of currency types to enhance currency management interactions.

### CODE REVIEW

File: Maple2.Server.Game/Commands/PlayerCommand.cs [hunk 1]
```diff
@@ -221,4 +223,46 @@ private void Handle(InvocationContext ctx, int points, short rank) {
...
+                    case "meret":
+                        session.Currency.GameMeret += amount;
+                        break;
+                    default:
+                        if (TryParseCurrencyType(currency, out CurrencyType parsedCurrencyType)) {
+                            session.Currency[parsedCurrencyType] += amount;
+                        } else {
+                            throw new ArgumentException("Invalid currency type.");
+                        }
+                        break;
```
Comment (Zintixx) [inline Maple2.Server.Game/Commands/PlayerCommand.cs:256]: I think we shouldn't return an exception if someone typos. It should just error to the player. We can also just remove the TryParseCurrencyType function as that's really just Enum.TryParse with more steps

```suggestion
                        if (!Enum.TryParse(currency, out CurrencyType currencyType)) {
                            ctx.Console.Error.WriteLine($"Failed to parse currency type: {currency}");
                            ctx.ExitCode = 1;
                            return;
                        }
                        session.Currency[currencyType] += amount;
                        break;
```

**Review by Zintixx — CHANGES_REQUESTED** [review]
GJ!!!

File: Maple2.Server.Game/Commands/PlayerCommand.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@
 using Maple2.Model.Metadata;
 using Maple2.Server.Game.Packets;
 using Maple2.Server.Game.Session;
+using Microsoft.EntityFrameworkCore.Metadata.Internal;
```
Comment (AngeloTadeucci) [inline Maple2.Server.Game/Commands/PlayerCommand.cs:10]: not needed

**Review by AngeloTadeucci — CHANGES_REQUESTED** [review]

**Review by AngeloTadeucci — APPROVED** [review]

File: Maple2.Server.Game/Commands/PlayerCommand.cs [hunk 3]
```diff
@@ -221,4 +222,44 @@ private void Handle(InvocationContext ctx, int points, short rank) {
...
+                        }
+                        session.Currency[currencyType] += amount;
+                        break;
+                }
+                ctx.ExitCode = 0;
+            } catch (SystemException ex) {
+                ctx.Console.Error.WriteLine(ex.Message);
+                ctx.ExitCode = 1;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (AngeloTadeucci) [reply Maple2.Server.Game/Commands/PlayerCommand.cs:263]: your refactor is the same exact code as before, stop cooking

[the remark this answers is not in the document]
Comment (Zintixx) [reply Maple2.Server.Game/Commands/PlayerCommand.cs:263]: This is not possible. GameMeret and Meso are not part of the CurrencyType enum.

[the remark this answers is not in the document]
Comment (Zintixx) [reply Maple2.Server.Game/Commands/PlayerCommand.cs:263]: @coderabbitai This is not possible. GameMeret and Meso are not part of the CurrencyType enum.

[the remark this answers is not in the document]
Comment (AngeloTadeucci) [reply Maple2.Server.Game/Commands/PlayerCommand.cs:263]: @coderabbitai your suggestion is valid but as Zin said, GameMeret and Meso are not part of CurrencyType enum

**Pull request merged**